includes/core/DataTransferObjects/class-data_transfer_object.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace WCPay\Core\DataTransferObjects;
+
+use ReflectionClass;
+use ReflectionException;
+
+abstract class Data_Transfer_Object
+{
+	/**
+	 * Return array of all properties.
+	 *
+	 * @return array
+	 */
+	public function to_array() {
+		$class = new ReflectionClass($this);
+		$data = [];
+		foreach ($class->getProperties() as $reflectionProperty) {
+			$data[$reflectionProperty->getName()] = $reflectionProperty->getValue();
+		}
+
+		return $data;
+	}
+
+	/**
+	 * Change DTO property value if property is not private.
+	 * @param ...$values
+	 * @return self Cloned object with changed properties.
+	 * @throws ReflectionException
+	 */
+	public function with(...$values)
```

Comment:
What would be the use case for this method?

Reply:
Something like this:

```
class Dto {
    private $id;
    private $amount;
	__construct($id, $amount) {
		$this->id = $id;
		$this->amount = $amount;
	}
}

$dto = new Dto(1, 100);
$dto->with(['id' => 2]);

echo $dto->id // will echo 2;
```

With this method, you could simply change any value in DTO.

Reply:
Would `fill` or `fill_with` be more descriptive method name?

Reply:
I remove this function since we allowed people to change data that needs to be immutable.